circuit/types/hint.go

```diff
@@ -8,13 +8,41 @@ import (
 	"github.com/ethereum/go-ethereum/crypto"
 )
 
+//func Keccak256(_ ecc.ID, inputs []*big.Int, outputs []*big.Int) error {
+//	var buf bytes.Buffer
+//	for i := 0; i < len(inputs); i++ {
+//		buf.Write(inputs[i].FillBytes(make([]byte, 32)))
+//	}
+//	hashVal := crypto.Keccak256Hash(buf.Bytes())
+//	result := outputs[0]
+//	result.SetBytes(hashVal[:])
+//	return nil
+//}
+
 func Keccak256(_ ecc.ID, inputs []*big.Int, outputs []*big.Int) error {
 	var buf bytes.Buffer
-	for i := 0; i < len(inputs); i++ {
+	for i := 0; i < 4; i++ {
 		buf.Write(inputs[i].FillBytes(make([]byte, 32)))
```

Comment:
shouldn't we add simple explanation comment for a new hashing scheme for the block commitment? like:
```go
// process blockNumber, createdAt, oldStateRoot, newStateRoot
for i := 0; i < 4; i++ {
	buf.Write(inputs[i].FillBytes(make([]byte, 32)))
}

// process pubData
for i := 4; i < len(inputs)-1; i = i + 8 {
	buf.Write(CovertBitsToBigInt(inputs[i : i+8]).FillBytes(make([]byte, 1)))
}

// process onChainOpsCount
buf.Write(inputs[len(inputs)-1].FillBytes(make([]byte, 32)))

```

Reply:
I will add some comments for this change.